lib/install/management/appliance.go

```diff
@@ -660,6 +739,132 @@ func (d *Dispatcher) createAppliance(conf *config.VirtualContainerHostConfigSpec
+			if soap.IsSoapFault(err) {
+
+				switch fault := soap.ToSoapFault(err).VimFault().(type) {
+				case types.AlreadyExists:
+					ok = true
+				case types.DuplicateName:
+					ok = true
+				case types.InvalidName:
+					d.op.Errorf("received InvalidName fault when attempting to create folder %s", fname)
+					return nil, fmt.Errorf("An invalid name was specified for the VCH: %s does not meet the naming criteria for a vch", vchName)
+				default:
+					d.op.Errorf("Encountered unexpected fault : %#v ", fault)
+					return nil, fmt.Errorf("unexpected fault when attempting to create the inventory folder %s please see vic-machine.log for more information", folderCreateTarget)
+				}
+
+				if ok {
+					// the folder already exists. Get the ref.
+					folder, err := d.session.Finder.Folder(d.op, folderCreateTarget)
+					if err != nil {
+						return nil, err
+					}
+
+					existingFolder = folder
+					continue
+				}
+			}
+
+			if soap.IsVimFault(err) {
+
+				switch fault := soap.ToVimFault(err).(type) {
+				case *types.AlreadyExists:
+					ok = true
+				case *types.DuplicateName:
+					ok = true
+				case *types.InvalidName:
+					d.op.Errorf("received InvalidName fault when attempting to create folder %s", fname)
+					return nil, fmt.Errorf("An invalid name was specified for the VCH: %s does not meet the naming criteria for a vch", vchName)
+				default:
+					d.op.Errorf("Encountered unexpected fault : %#v ", fault)
+					return nil, fmt.Errorf("unexpected fault when attempting to create the inventory folder %s please see vic-machine.log for more information", folderCreateTarget)
+				}
+
+				if ok {
+					// the folder already exists. Get the ref.
+					folder, err := d.session.Finder.Folder(d.op, folderCreateTarget)
+					if err != nil {
+						return nil, err
+					}
+
+					existingFolder = folder
+					continue
+				}
+			}
+
+			if err != nil {
+				// finally check for a task error
+				switch unknownError := err.(type) {
+				case task.Error:
+					switch fault := unknownError.Fault().(type) {
+					case *types.AlreadyExists:
+						ok = true
+					case *types.DuplicateName:
+						ok = true
+					case *types.InvalidName:
+						d.op.Errorf("received InvalidName fault when attempting to create folder %s", fname)
+						return nil, fmt.Errorf("An invalid name was specified for the VCH: %s does not meet the naming criteria for a vch", vchName)
+					default:
+						d.op.Errorf("Encountered unexpected fault : %#v ", fault)
+						return nil, fmt.Errorf("unexpected fault when attempting to create the inventory folder %s please see vic-machine.log for more information", folderCreateTarget)
+					}
+				default:
+					d.op.Errorf("Encountered unexpected error : %#v ", unknownError)
+					return nil, fmt.Errorf("unexpected error when attempting to create the inventory folder %s please see vic-machine.log for more information", folderCreateTarget)
+				}
+
+				if ok {
+					// the folder already exists. Get the ref.
```

Comment:
change this comment, `AlreadyExists` style fault is actually bad. Something else like "the folder is already in the inventory".

Reply:
removed comment altogether.